x-pack/platform/plugins/shared/lens/public/react_embeddable/user_messages/api.test.ts

```diff
@@ -0,0 +1,277 @@
+      }
+    });
+
+    it('should return basic validation for missing parts of the config', () => {
+      const { userMessagesApi, internalApi } = buildUserMessagesApi();
+      // no doc scenario
+      internalApi.updateVisualizationContext({
+        ...internalApi.getVisualizationContext(),
+        activeAttributes: undefined,
+      });
+      for (const locationId of ALL_LOCATIONS) {
+        expect(userMessagesApi.getUserMessages(locationId).map(({ uniqueId }) => uniqueId)).toEqual(
+          ['editor_missing_vis_type', 'editor_unknown_datasource_type']
+        );
+      }
+    });
+
+    it('should detect a URL conflict', () => {
+      const { userMessagesApi } = buildUserMessagesApi({ outcome: 'conflict' });
+
+      for (const locationId of ALL_LOCATIONS.filter((id) => id !== 'visualization')) {
+        expect(userMessagesApi.getUserMessages(locationId)).toEqual([]);
+      }
+      expect(userMessagesApi.getUserMessages('visualization')).toEqual(
+        expect.arrayContaining([expect.objectContaining({ uniqueId: 'url-conflict' })])
+      );
+    });
+
+    it('should filter messages based on severity criteria', () => {
+      const { userMessagesApi } = buildUserMessagesApi();
+      const userMessageError = createUserMessage();
+      const userMessageWarning = createUserMessage(['embeddableBadge'], 'warning');
+      const userMessageInfo = createUserMessage(['embeddableBadge'], 'info');
+      userMessagesApi.addUserMessages([userMessageError, userMessageWarning, userMessageInfo]);
+      expect(userMessagesApi.getUserMessages('embeddableBadge', { severity: 'error' })).toEqual(
+        expect.arrayContaining([userMessageError])
+      );
+      expect(userMessagesApi.getUserMessages('embeddableBadge', { severity: 'warning' })).toEqual(
+        expect.arrayContaining([userMessageWarning])
+      );
+      expect(userMessagesApi.getUserMessages('embeddableBadge', { severity: 'info' })).toEqual(
+        expect.arrayContaining([userMessageInfo])
+      );
+    });
+
+    it('should filter messages based on locationId', () => {
+      const { userMessagesApi } = buildUserMessagesApi();
+      const userMessageEmbeddable = createUserMessage(['embeddableBadge']);
+      const userMessageVisualization = createUserMessage(['visualization']);
+      const userMessageEmbeddableVisualization = createUserMessage([
+        'visualization',
+        'embeddableBadge',
+      ]);
+      userMessagesApi.addUserMessages([
+        userMessageEmbeddable,
+        userMessageVisualization,
+        userMessageEmbeddableVisualization,
+      ]);
+      expect(userMessagesApi.getUserMessages('embeddableBadge').length).toEqual(2);
+      expect(userMessagesApi.getUserMessages('visualization').length).toEqual(2);
+      expect(userMessagesApi.getUserMessages('visualizationOnEmbeddable').length).toEqual(0);
+    });
+
+    it('should return deeper validation messages from both datasource and visualization', () => {});
```

Comment:
you can probably skip or remove this for now if you are not going to implement it now